server/cmwell-ws/app/controllers/SpHandler.scala

```diff
@@ -89,7 +89,7 @@ class SpHandlerController @Inject()(crudServiceFS: CRUDServiceFS)(implicit ec: E
     import SpHandler._
 
     Try(parse(req)) match {
-      case Failure(_) => Future.successful(wsutil.exceptionToResponse(new IllegalArgumentException("Parsing error")))
+      case Failure(_) => Future.successful(wsutil.exceptionToResponse(new IllegalArgumentException("SpHandler Parsing error")))
```

Comment:
Exposing internal names of classes as an Exception that the user see (it will be converted to an HTTP 400) is not a good practice in terms of security. I'd suggest not to change "SpHandler Parsing error" to "_sp API: parsing error" or perhaps just leave it as it is today (I know you're matching against it in STP, but why not match to "Parsing error"?)

Reply:
> or perhaps just leave it as it is today

I'll work on proper retry mechanism later - it needs a lot of attention.
[shouldRetry](https://github.com/CM-Well/CM-Well/blob/master/server/cmwell-sparql-agent/src/main/scala/cmwell/tools/data/sparql/SparqlProcessorManager.scala#L608) was commented in SparqlProcessorManager.

I could refactor SparqlProcessorManager shouldRetry and Retry later - i don't think it's a priority right now.